common/lib/common-definitions/src/logger.ts

```diff
@@ -45,12 +45,166 @@ export interface ITelemetryBaseEvent extends ITelemetryProperties {
+     */
+    apiName: string;
+    /*
+     * Duration of the call.
+     */
+    duration: number;
+    /*
+     * Identifier of a request call instance.
+     * Used for correlation (TODO) between request and other telemetry items.
+     */
+    id: string;
+    /*
+     * Flag indicating if call suceeded or failed.
+     */
+    success: boolean;
+    /*
+     * Optional response status.
+     */
+    status?: string;
+    /*
+     * JSON stringifed bag od details on the event.
+     */
+    details?: string;
+}
+
+/**
+ * General-use Error event. Typically represents an exception that causes an operation to fail.
+ */
+export interface ITelemetryGenErrorEvent extends ITelemetryGenEventBase {
+    type: "error";
+    /*
+     * Used for exceptions grouping (ex exception type + function).
+     */
+    errorCode: string;
+    /*
+     * Optional message.
+     */
+    message?: string;
+    /*
+     * Severity level.
+     */
+    severityLevel: TelemetryErrorCategory;
+    /*
+     * JSON stringifed stack trace
+     */
+    stackTrace?: string;
+    /*
+     * JSON stringifed details on condition(s) leading to this error
+     */
+    details?: string;
+}
+
+/**
+ * General-use Service event. Represents a call from FF to an external service or storage.
+ */
+export interface ITelemetryGenServiceEvent extends ITelemetryGenEventBase {
+    type: "service";
+    /*
+     * Duration of the call.
+     */
+    duration: number;
+    /*
+     * Identifier of a dependency call instance (correlation id).
+     */
+    id?: string;
+    /*
+     * Target site of a dependency call. Examples are server name, host address (with PII considerations).
+     */
+    target: string;
+    /*
+     * Result code. Ex. HTTP status code.
+     */
+    resultCode: string;
+    /*
+     * Flag indicating success/failure.
+     */
+    success: boolean;
+}
+
+/**
+ * General-use class event - Represents an event firing on a specific object.
+ * (Need a better name. "Event" has few different meanings in logger.)
+ */
+export interface ITelemetryGenClassEvent extends ITelemetryGenEventBase {
+    type: "event";
+    /*
+     * JSON stringifed bag od details on the event.
+     */
+    details?: string;
+}
+
+export type ITelemetryGenEvent =
+    | ITelemetryGenApiEvent
+    | ITelemetryGenServiceEvent
+    | ITelemetryGenErrorEvent
+    | ITelemetryGenClassEvent;
+
 /**
  * Interface to output telemetry events.
  * Implemented by hosting app / loader
  */
 export interface ITelemetryBaseLogger {
+    /**
+     * Unstructured dev-only telemetry. Event properties can change across minor or major releases.
+     * This stream is inteded to generate telemetry that is actionable to Fluid developers.
+     * Implemented by derived classes
+     * @param event - Telemetry event to send over
+     */
     send(event: ITelemetryBaseEvent): void;
+
+    /**
+     * General-use, static telemetry that is meant for general consumption. Events are strongly typed and follow
```

Comment:
Few more notes here. We should definitely consider various pivots and constraints. With current approach we are really offering this:

- DebugLogs (current logs)
- API telemetry.
- Service Calls telemetry
- API Requests telemetry.
-  Errors.

This more in line with how app insights may model telemetry: https://learn.microsoft.com/en-us/azure/azure-monitor/app/data-model

Other approach would be more traditional log-level approach:
debug:  Internal logs (current telemetry)
log:  inform your user about things they may need to know.
info: inform the user about something that they need to take a concrete action on.
warn: feature may stop functioning correctly
error
(an example)